src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -267,18 +374,46 @@ static void OnResolve(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t inter
     }
 }
 
+static CHIP_ERROR ResolveWithContext(ResolveContext * sdCtx, uint32_t interfaceId, const char * type, const char * name,
+                                     const char * domain, ResolveContextWithType * contextWithType)
+{
+    auto sdRef = sdCtx->serviceRef; // Mandatory copy because of kDNSServiceFlagsShareConnection
+
+    auto err = DNSServiceResolve(&sdRef, kResolveFlags, interfaceId, name, type, domain, OnResolve, contextWithType);
+    VerifyOrReturnError(kDNSServiceErr_NoError == err, sdCtx->Finalize(err));
+
+    // Set the flag to start the timer for resolve on SRP domain to complete if a resolve has been requested on the SRP domain.
```

Comment:
So this is a little silly, in that we set it true here but then one of the places where we call it immediately sets it false...

We should just set it false if we are starting one resolve, set it true if we are starting two resolves, at the callsites of this function.

Reply:
that was silly. fixed it.